mlir/utils/jenkins/Jenkinsfile

```diff
@@ -1057,5 +1058,68 @@ pipeline {
                 }
             }
         }
+
+        stage ("Code coverage") {
+            when {
+                beforeAgent true;
+                equals expected: true, actual: params.runCodeCoverage;
+            }
+            matrix {
+                axes {
+                    axis {
+                        name 'CHIP'
+                        values 'gfx908', 'gfx90a'    //, 'gfx1030', 'gfx1100'
+                    }
+                }
+                agent {
+                    docker {
+                        image dockerImage()
+                        args dockerArgs()
+                        label getLabelFromChip("${CHIP}")
+                        alwaysPull true
+                    }
+                }
+                environment {
+                    PATH="/opt/rocm/llvm/bin:$PATH"
+                    HOME="${WORKSPACE}"
+                    // Note the %m to avoid issues with threads and dynamic libraries.
+                    LLVM_PROFILE_FILE="${WORKSPACE}/build/%m-%p.profraw"
+                }
+                stages {
+                    stage ("body") {
+                        steps {
+                            // Build with profiling on, and just code-generation tests.
+                            sh 'rm -f build/CMakeCache.txt'
+                            sh 'rm -f build/*.profraw'
+                            buildProject('check-rocmlir-build-only',
+                                         '-DBUILD_FAT_LIBROCKCOMPILER=ON -DCMAKE_BUILD_TYPE=debug -DCMAKE_CXX_FLAGS_DEBUG="-O0 -fprofile-instr-generate -fcoverage-mapping"')
```

Comment:
Check if there's something in the LLVM CMake files for this

Reply:
There's -DLLVM_BUILD_INSTRUMENTED_COVERAGE, though it doesn't include the -O0.  Is that the sort of thing you had in mind?

Reply:
Yeah, though I'm not convinced we *need* the `-O0`

Or maybe we'd get it by just doing a `Debug` build too